src/utils/notifications.ts

```diff
@@ -56,3 +58,28 @@ export function localNotificationsAreSilenced(cli: MatrixClient): boolean {
     const event = cli.getAccountData(eventType);
     return event?.getContent<LocalNotificationSettings>()?.is_silenced ?? false;
 }
+
+export function clearAllNotifications(client: MatrixClient): Promise<Array<{}>> {
+    const receiptPromises = client.getRooms().reduce((promises, room: Room) => {
+        if (room.getUnreadNotificationCount() > 0) {
+            const roomEvents = room.getLiveTimeline().getEvents();
+            const lastThreadEvents = room.lastThread?.events;
+
+            const lastRoomEvent = roomEvents?.[roomEvents?.length - 1];
+            const lastThreadLastEvent = lastThreadEvents?.[lastThreadEvents?.length - 1];
+
+            const lastEvent = (lastRoomEvent?.getTs() ?? 0) > (lastThreadLastEvent?.getTs() ?? 0)
+                ? lastRoomEvent
+                : lastThreadLastEvent;
+
+            if (lastEvent) {
+                const promise = client.sendReadReceipt(lastEvent, ReceiptType.Read, true);
```

Comment:
What about private read receipts?

Reply:
@SimonBrandner I am unsure how private read receipts are supposed to work. 

I have copied the logic that is in `TimelinePanel` that checks the `sendReadReceipts` setting per room to decide to send `ReceiptType.Read` if `true` and `ReceiptType.ReadPrivate` if `false`. 

Can you confirm this is correct? 
By the way, the clear notification feature did not support private read receipts before this pull request.